noir/noir-repo/compiler/noirc_driver/src/lib.rs

```diff
@@ -124,6 +124,12 @@ pub struct CompileOptions {
     /// This check should always be run on production code.
     #[arg(long)]
     pub skip_underconstrained_check: bool,
+
+    /// Setting to decide on an inlining strategy for brillig functions.
+    /// A more aggressive inliner should generate larger programs but more optimized
+    /// A less aggressive inliner should generate smaller programs
+    #[arg(long, hide = true, allow_hyphen_values = true, default_value_t = f64::INFINITY)]
+    pub inliner_aggressiveness: f64,
```

Comment:
What's the reasoning behind making this an `f64`? I'd like to avoid using floats if possible.

Reply:
I think it's a good usecase for floating point numbers: we don't want to handle overflows, we want to be able to express infinity and negative infinity, we don't care about precision since it's an approximation...

Reply:
We can probably use i64 and we'd probably never overflow though, but users would need to pass stuff like -9223372036854775808  for never inline

Reply:
`i64::MIN` seems fine to pass. I don't feel super strongly on avoiding float though so I defer to @TomAFrench here.

Reply:
Changed it to i64!